rfcs/IncrementalDelivery.md

```diff
@@ -6,9 +6,15 @@ This RFC proposes adding a specification of how "Incremental" GraphQL results sh
 
 An Incremental result is a GraphQL result that is split up into multiple payloads, allowing clients quicker access to parts of the results. Currently this is supported by the proposed `@defer` and `@stream` directives ([RFC](https://github.com/graphql/graphql-spec/blob/master/rfcs/DeferStream.md)).
 
-## `transfer-encoding: chunked`
+### HTTP/1.1
 
-The HTTP response for an incrementally delivered response should contain the `transfer-encoding: chunked` response header. Chunked transfer encoding allows the body of the response to be delivered as a series of chunks, allowing clients to read each chunk of the response as it is sent without waiting for the entire response.
+An incrementally delivered response should contain the `transfer-encoding: chunked` response header when using HTTP/1.1. Chunked transfer encoding allows the body of the response to be delivered as a series of chunks, allowing clients to read each chunk of the response as it is sent without waiting for the entire response.
+
+### HTTP/2
+
+Because of improved data streaming mechanisms, HTTP/2 prohibits the use of the `transfer-encoding` header. It is very likely that compliant servers will treat requests containing the header as malformed ([see section 8.2.2. Connection-Specific Header Fields in HTTP/2 spec](https://datatracker.ietf.org/doc/html/rfc9113#section-8.1)).
+
+Having said this, HTTP/2 enabled servers must not set the `transfer-encoding` header during response.
```

Comment:
This seems redundant with the first sentence of the above paragraph.

> HTTP/2 prohibits the use of the `transfer-encoding` header.

Reply:
The paragraph above is an explanation/description and this is the spec requirement. I added it for brevity.

Reply:
I like the idea, thanks! https://github.com/graphql/graphql-over-http/pull/252/commits/b3ebf9f47fb3e2bf7bfdd9352cbeb58cbe54792e